meteor/server/api/rest/rest.ts

```diff
@@ -37,7 +37,7 @@ function typeConvertUrlParameters(args: any[]) {
 			if (!_.isNaN(Number(val))) {
 				val = Number(val)
 			} else {
-				let json: any = null
+				let json: object = {}
```

Comment:
Is it necessary to change it to an object? If so, then the condition after the try-catch becomes a tautology – under the assumption that all values parsed takes on the form of an arbitrary object. Thus you can change this to:
```typescript
try {
  val = JSON.parse(val)
} catch (e) {
  val = {}
}
```
And if the check is still needed (in case of failure or fals-ish values), then this can become:
```typescript
try {
  const data = JSON.parse(val)
  val = data ? data : val
} catch (e) {
  // ignore
}
```
This can be simplified a bit if we assume that parsed values are only of the type object.

But if the assumption is incorrect, then when you get the parsed JSON object, this could be `null`, `undefined`, `false`, `[]` or some other non-object value. So if it is important to enforce that it is an object, then perhaps we should check for it?

Reply:
I have added the first suggestion :-)